packages/backend/src/managers/playground/McpServerManager.ts

```diff
@@ -15,52 +15,67 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
-import type { Disposable } from '@podman-desktop/api';
-import { promises } from 'node:fs';
 import path from 'node:path';
-import { type McpSettings, type McpServer, McpServerType } from '../../models/mcpTypes';
+import { type Disposable, EventEmitter } from '@podman-desktop/api';
+import { MSG_MCP_SERVERS_UPDATE } from '@shared/Messages';
+import { type McpSettings, McpServerType, type McpClient } from '@shared/models/McpSettings';
+import type { RpcExtension } from '@shared/messages/MessageProxy';
+import { JsonWatcher } from '../../utils/JsonWatcher';
+import { Publisher } from '../../utils/Publisher';
+import { toMcpClients } from '../../utils/mcpUtils';
 
 // TODO: Agree on the name of the file and its location
 const MCP_SETTINGS = 'mcp-settings.json';
 
-// TODO: Consider using JsonWatcher in case the file is going to be updated externally
-// Depending on the use case, we might want to watch for changes in the settings file.
-// 1. As a user, I want to be able to modify the mcp-settings.json file and have the changes reflected in the application without restarting it.
-// 2. As a user, I want to be able to add/remove servers from the mcp-settings.json while also having the application to provide a UI to manage the servers.
-// 3. As a user, I don't want to tinker with the mcp-settings.json file and would like to have the application to provide a UI to manage the servers.
-// Cases 1 and 2 are covered by JsonWatcher, but case 3 is not.
-// For 2 and 3, we need to implement a UI to manage the servers.
-export class McpServerManager implements Disposable {
+export class McpServerManager extends Publisher<McpSettings> implements Disposable {
   private readonly settingsFile: string;
+  private mcpSettings: McpSettings;
+  private readonly _onUpdate: EventEmitter<McpSettings>;
```

Comment:
The `_onUpdate` do not have any listeners, you are firing events, but I don't anything subscribing to it.

Reply:
:+1: This is part of the upcoming feature. Removing it in scope of this PR